packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -570,6 +595,26 @@ export class Function extends FunctionBase {
     }
 
     this.currentVersionOptions = props.currentVersionOptions;
+
+    if (props.filesystems) {
+      // max 1 filesystem allowed
```

Comment:
This is unnecessary future proofing.. If only a single file system is allowed, then why do we need the property to be `filesystems`? Shall we change it to `filesystem`? In the future if we want to add additional filesystems, we can either deprecate the singular property or add `additionalFileSystems`.

Reply:
OK. Let me revise it.

Reply:
Hi @eladb 

It's `filesystem` now.

And, to even scope down the required iam policies, I changed `managedPolicies` to `policies` and bake the required policy statements like this in the `fromEfsAccessPoint`

```ts
      policies: [
        new iam.PolicyStatement({
          actions: [ 'elasticfilesystem:ClientMount' ],
          resources: [ '*' ],
          conditions: {
            StringEquals: {
              'elasticfilesystem:AccessPointArn': ap.accessPointArn,
            },
          },
        }),
        new iam.PolicyStatement({
          actions: ['elasticfilesystem:ClientWrite'],
          resources: [ Stack.of(ap).formatArn({
            service: 'elasticfilesystem',
            resource: 'file-system',
            resourceName: ap.fileSystem.fileSystemId,
          }) ],
        }),
      ],
```

Let me know if it looks better now.